### PR TITLE
Implement pagination on get_content_metadata

### PR DESCRIPTION
**Pull request by bbaker6225**
Update the get_content_metadata endpoint on enterprise catalogs to
paginate the results (as opposed to just returning them all on one
page).
Consumers also have the option to get all of the metadata on a single
page by adding the `?traverse_pagination` query parameter
* Update devstack client id & secret
* Add utility to get the content metadata associated with a catalog
* Small naming,test fixes

ENT-2802

**Testing Instructions**
* Hit `api/v1/{catalog_uuid}/refresh_metadata` to associate content metadata with some catalog
* Change your local setting for `REST_FRAMEWORK.PAGE_SIZE` to `1` or something low enough to force pagination
* Hit `api/v1/{catalog_uuid}/get_content_metadata.json` and see the results paginated
* Hit `api/v1/{catalog_uuid}/get_content_metadata.json/?traverse_pagination=1` and see the results all placed onto one page

### CODE REVIEW

File: enterprise_catalog/apps/api_client/discovery.py [hunk 1]
```diff
@@ -31,33 +31,7 @@ def oauth2_client_id(self):
     def oauth2_client_secret(self):
         return settings.BACKEND_SERVICE_EDX_OAUTH2_SECRET
 
-    def traverse_pagination(self, response, content_filter_query, query_params):
```
Comment (bbaker6225) [inline enterprise_catalog/apps/api_client/discovery.py:34]: This is not needed as part of the discovery client as we don't want to page through discovery when the consumer pages through our API

File: enterprise_catalog/apps/catalog/models.py [hunk 2]
```diff
@@ -132,6 +132,18 @@ def __str__(self):
...
+    @property
+    def content_metadata(self):
+        """
+        Helper to retrieve the content metadata associated with the catalog.
+
+        Returns:
+            Queryset: The queryset of associated content metadata
+        """
+        if not self.catalog_query:
+            return ContentMetadata.objects.none()
```
Comment (bbaker6225) [inline enterprise_catalog/apps/catalog/models.py:144]: Kind of silly, but ensures we always return a `Queryset` and keeps things nice elsewhere

File: enterprise_catalog/settings/devstack.py [hunk 3]
```diff
@@ -14,8 +14,8 @@
 )
 
 # OAuth2 variables specific to backend service API calls.
-BACKEND_SERVICE_EDX_OAUTH2_KEY = os.environ.get('BACKEND_SERVICE_EDX_OAUTH2_KEY', 'enterprise_catalog-key')
-BACKEND_SERVICE_EDX_OAUTH2_SECRET = os.environ.get('BACKEND_SERVICE_EDX_OAUTH2_SECRET', 'enterprise_catalog-secret')
+BACKEND_SERVICE_EDX_OAUTH2_KEY = os.environ.get('BACKEND_SERVICE_EDX_OAUTH2_KEY', 'enterprise-catalog-backend-service-key')
```
Comment (bbaker6225) [inline enterprise_catalog/settings/devstack.py:17]: Fixing to match what's in local lms

File: enterprise_catalog/apps/api/v1/serializers.py [hunk 4]
```diff
@@ -92,3 +92,28 @@ class EnterpriseCatalogCreateSerializer(EnterpriseCatalogSerializer):
...
+        """
+
+
+class ContentMetadataSerializer(ImmutableStateSerializer):
+    """
+    Serializer for rendering Content Metadata objects
+    """
+
+    def to_representation(self, instance):
+        return instance.json_metadata
```
Comment (iloveagent57) [inline enterprise_catalog/apps/api/v1/serializers.py:119]: Should we actually return a copy of this?

File: enterprise_catalog/apps/api/v1/serializers.py [hunk 5]
```diff
@@ -92,3 +92,28 @@ class EnterpriseCatalogCreateSerializer(EnterpriseCatalogSerializer):
     UUID is writable to allow importing existing Enterprise Catalogs and keeping the same UUID
     """
     uuid = serializers.UUIDField(read_only=False, required=False)
+
+
+class ImmutableStateSerializer(serializers.Serializer):
```
Comment (bbaker6225) [inline enterprise_catalog/apps/api/v1/serializers.py:97]: Copied from https://github.com/edx/enterprise-catalog/pull/95.
The rest of that PR is omitted as it seems easier to add on to the `ContentMetadataSerializer` as opposed to readjusting the views again

File: enterprise_catalog/apps/api/v1/tests/test_views.py [hunk 6]
```diff
@@ -452,21 +465,47 @@ def test_get_content_metadata(self):
...
-        metadata_2 = ContentMetadataFactory(json_metadata=json_metadata_2, content_key='second')
-        self.add_metadata_to_catalog(self.enterprise_catalog, [metadata_1, metadata_2])
-
+        # Create enough metadata to force pagination
+        metadata = ContentMetadataFactory.create_batch(api_settings.PAGE_SIZE + 1)
+        self.add_metadata_to_catalog(self.enterprise_catalog, metadata)
         url = self._get_content_metadata_url(self.enterprise_catalog)
         response = self.client.get(url)
         self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual((response.json()['count']), api_settings.PAGE_SIZE + 1)
```
Comment (iloveagent57) [inline enterprise_catalog/apps/api/v1/tests/test_views.py]: maybe do `response_data = response.json()` and use that variable in all of these lines?

File: enterprise_catalog/apps/api/v1/views.py [hunk 7]
```diff
@@ -106,32 +107,84 @@ def contains_content_items(self, request, uuid, course_run_ids, program_uuids, *
...
+            response (HttpResponse): The existing DRF response to add on to
+
+        Returns:
+            HttpResponse: The new response with additional fields added on
+        """
+        enterprise_catalog = self.get_enterprise_catalog()
+        response.data['uuid'] = enterprise_catalog.uuid
+        response.data['title'] = enterprise_catalog.title
+        response.data['enterprise_customer'] = enterprise_catalog.enterprise_uuid
+        response.data.move_to_end('results')  # Place the results at the end of the response again
```
Comment (iloveagent57) [inline enterprise_catalog/apps/api/v1/views.py:160]: :) nice!

**Review by iloveagent57 — COMMENTED** [review]
2 real minor suggestions, looks good!

File: enterprise_catalog/apps/api/v1/serializers.py (hunk 4 above)
Comment (bbaker6225) [reply enterprise_catalog/apps/api/v1/serializers.py:119]: We're not making any edits to the metadata, so it should be fine to just return it for rendering

File: enterprise_catalog/apps/api/v1/tests/test_views.py (hunk 6 above)
Comment (bbaker6225) [reply enterprise_catalog/apps/api/v1/tests/test_views.py]: :+1:

**Review by chavesj — APPROVED** [review]
LGTM 🚢

**Pull request merged**